modules/account-lib/src/coin/hbar/keyPair.ts

```diff
@@ -52,8 +54,11 @@ export class KeyPair extends Ed25519KeyPair {
 
   /** @inheritdoc */
   recordKeysFromPrivateKeyInProtocolFormat(prv: string): DefaultKeys {
+    if (prv.length % 2 !== 0 && /^([a-f0-9]{2})+$/i.test(prv)) {
+      throw new InvalidKey('Invalid private key length. Must be a hex and multiple of 2');
+    }

```

Comment:
There's some lack of understanding here - the length check is now redundant, since the regex only matches on pairs of hex digits. We're also missing a negation on the regex test so throwing the `InvalidKey` error isn't logically possible.
```suggestion
    if (!/^([a-f0-9]{2})+$/i.test(prv)) {
      throw new InvalidKey('Invalid private key. Must be a hex string with even length');
    }

```

Please write a unit test for these cases, since this is apparently uncovered.